easy_my_coop_website_portal/controllers/main.py

```diff
@@ -0,0 +1,187 @@
+            page=page,
+            step=self._items_per_page
+        )
+        # content according to pager and archive selected
+        invoices = invoice_mgr.search(
+            domain, limit=self._items_per_page, offset=pager['offset'])
+        values.update({
+            'date': date_begin,
+            'capital_requests': invoices,
+            'page_name': 'invoice',
+            'pager': pager,
+            'archive_groups': archive_groups,
+            'default_url': '/my/release_capital_request',
+        })
+        return request.website.render(
+            "easy_my_coop_website_portal.portal_my_capital_releases",
+            values
+        )
+
+    @http.route(['/my/release_capital_request/pdf/<int:oid>'],
+                type='http', auth="user", website=True)
+    def get_release_capital_request(self, oid=-1, **kw):
+        """Render the pdf of the given release capital request"""
+        # Get the release capital request and raise an error if the user
+        # is not allowed to access to it or if the object is not found.
+        partner = request.env.user.partner_id
+        invoice_mgr = request.env['account.invoice']
+        capital_request = invoice_mgr.sudo().browse(oid)
+        try:
+            if capital_request.partner_id != partner:
+                raise Forbidden()
+        except AccessError:
+            raise Forbidden()
+        except MissingError:
+            raise NotFound()
+        # Get the pdf
+        report_mgr = request.env['report'].sudo()
+        pdf = report_mgr.get_pdf(
+            capital_request,
+            'easy_my_coop.theme_invoice_G002'
+        )
+        filename = "Release Capital Request - %d" % capital_request.id
+        return self._render_pdf(pdf, filename)
+
+    @http.route(['/my/cooperator_certificate/pdf'],
+                type='http', auth="user", website=True)
+    def get_cooperator_certificat(self, **kw):
+        """Render the cooperator certificate pdf of the current user"""
+        partner = request.env.user.partner_id
+        report_mgr = request.env['report'].sudo()
+        pdf = report_mgr.get_pdf(
+            partner,
+            'easy_my_coop.cooperator_certificat_G001'
+        )
+        filename = "Cooperator Certificate - %s" % partner.name
+        return self._render_pdf(pdf, filename)
+
+    def _render_pdf(self, pdf, filename):
+        """Render a http response for a pdf"""
+        pdfhttpheaders = [
+            ('Content-Disposition', 'inline; filename="%s.pdf"' % filename),
+            ('Content-Type', 'application/pdf'),
+            ('Content-Length', len(pdf))
+        ]
+        return request.make_response(pdf, headers=pdfhttpheaders)
+
+    def _get_archive_groups_sudo(self, model, domain=None, fields=None,
+                                 groupby="create_date",
+                                 order="create_date desc"):
+        """Same as the one from website_portal_v10 except that it runs
```

Comment:
not possible to call the _get_archive_groups of website_portal_v10 with sudo() instruction?
something like _get_archive_groups.sudo() ?

Reply:
I have tried at the beginning but this function comes from a controller which is not part of a `recordset` and therefor has no member `sudo()`. I have tried to change the become `sudo` by changing the user in the context but the current context but it doesn't work either. If someone has, a better way to do this, I will take it. But for now it's the only way that works.

Reply:
ok